database/kwdb.sql

```diff
@@ -0,0 +1,222 @@
+PRAGMA foreign_keys = 1;
+BEGIN TRANSACTION;
+-- Tables
+
+-- This table holds the metadata of the config files handled by kw
+CREATE TABLE IF NOT EXISTS "config" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "name" TEXT NOT NULL UNIQUE,
+  "description" TEXT,
+  "path" TEXT NOT NULL UNIQUE,
+  PRIMARY KEY("id")
+);
+
+-- This table holds the names of the commands that kw keeps track of, like
+-- build and deploy
+CREATE TABLE IF NOT EXISTS "command_label" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "name" TEXT NOT NULL UNIQUE,
+  PRIMARY KEY("id")
+);
+
+-- Table containing the possible exit status of an executed commmand
+CREATE TABLE IF NOT EXISTS "status" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "name" TEXT NOT NULL UNIQUE,
+  PRIMARY KEY("id")
+);
+
+-- Table containing user created tags
+CREATE TABLE IF NOT EXISTS "tag" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "name" TEXT NOT NULL UNIQUE,
+  "active" INTEGER NOT NULL CHECK ("active" IN (0, 1)) DEFAULT 1,
+  PRIMARY KEY("id")
+);
+
+-- This is the table that holds the events triggered by kw, currently pertains
+-- to the executed commands that are saved and the pomodoro sessions created
+CREATE TABLE IF NOT EXISTS "event" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "date" TEXT NOT NULL,
+  "time" TEXT,
+  PRIMARY KEY("id")
+);
+
+-- This is the relationship between an "event" that executes a given
+-- "command_label"
+CREATE TABLE IF NOT EXISTS "executed" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "command_label_id" INTEGER NOT NULL,
+  "elapsed_time_in_secs" INTEGER NOT NULL,
+  "status_id" INTEGER NOT NULL,
+  PRIMARY KEY("id"),
+  FOREIGN KEY("id") REFERENCES "event"("id")
+    ON UPDATE CASCADE ON DELETE CASCADE,
+  FOREIGN KEY("command_label_id") REFERENCES "command_label"("id")
+    ON UPDATE CASCADE ON DELETE RESTRICT,
+  FOREIGN KEY("status_id") REFERENCES "status"("id")
+    ON UPDATE CASCADE ON DELETE RESTRICT
+);
+
+-- Table containing the information related to each pomodoro timebox
+CREATE TABLE IF NOT EXISTS "timebox" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "duration" INTEGER NOT NULL,
+  "description" TEXT,
+  "tag_id" INTEGER NOT NULL,
+  PRIMARY KEY("id"),
+  FOREIGN KEY("id") REFERENCES "event"("id")
+    ON UPDATE CASCADE ON DELETE CASCADE,
+  FOREIGN KEY("tag_id") REFERENCES "tag"("id")
+    ON UPDATE CASCADE ON DELETE RESTRICT
+);
+
+-- Populate commands and status tables
+INSERT OR IGNORE INTO "status" ("name") VALUES
+  ('success'),
+  ('failure'),
+  ('interrupted'),
+  ('unknown');
+
+INSERT OR IGNORE INTO "command_label" ("name") VALUES
+  ('build'),
+  ('codestyle'),
+  ('configm'),
+  ('deploy'),
+  ('device'),
+  ('drm'),
+  ('explore'),
+  ('list'),
+  ('maintainers'),
+  ('modules_deploy'),
+  ('ssh'),
+  ('uninstall');
```

Comment:
Please, update this list with all the command options (e.g., VM, self-update, etc).